ConcernsCaseWork/ConcernsCaseWork.API/Features/Decision/Outcome/Create.cs

```diff
@@ -0,0 +1,171 @@
+					.ThenInclude(x => x.BusinessAreasConsulted)
+					.FirstOrDefaultAsync(o => o.Urn == request.ConcernsCaseUrn);
+
+				if (concernCase == null)
+				{
+					throw new NotFoundException($"Not Found: Concern with id {request.ConcernsCaseUrn}");
+				}
+
+				var decision = concernCase.Decisions.FirstOrDefault(d => d.DecisionId == request.DecisionId);
+
+				if (decision == null)
+				{
+					throw new NotFoundException($"Not Found: Decision with id {request.DecisionId}, Case {request.ConcernsCaseUrn}");
+				}
+
+				if (decision.Outcome != null)
+				{
+					throw new ResourceConflictException($"Conflict: Decision with id {request.DecisionId} already has an outcome, Case {request.ConcernsCaseUrn}");
+				}
+
+				var now = DateTime.Now;
+				var outcome = new DecisionOutcome()
+				{
+					DecisionId = request.DecisionId,
+					Status = request.Model.Status,
+					TotalAmount = request.Model.TotalAmount,
+					DecisionMadeDate = request.Model.DecisionMadeDate,
+					DecisionEffectiveFromDate = request.Model.DecisionEffectiveFromDate,
+					Authorizer = request.Model.Authorizer,
+					BusinessAreasConsulted = request.Model.BusinessAreasConsulted.Select(b => new DecisionOutcomeBusinessAreaMapping()
+					{
+						DecisionOutcomeBusinessId = b
+					}).ToList(),
+					CreatedAt = now,
+					UpdatedAt = now
+				};
+
+				var result = _context.DecisionOutcomes.Add(outcome);
+				await _context.SaveChangesAsync(cancellationToken);
+
+				var decisionCreatedNotification = new DecisionOutcomeCreatedNotification() { Id = result.Entity.DecisionOutcomeId, CaseId = request.ConcernsCaseUrn };
+				await _mediator.Publish(decisionCreatedNotification);
+
+				return new CommandResult()
+				{
+					ConcernsCaseUrn = request.ConcernsCaseUrn,
+					DecisionId = request.DecisionId,
+					DecisionOutcomeId = result.Entity.DecisionOutcomeId,
+					Outcome = result.Entity
+				};
+			}
+		}
+
+		//ToDo: BB 25/07/2023 Once we have more events decide on specific folder structure to organise events and handles
+		public class DecisionOutcomeCreatedNotification : INotification
+		{
+			public int Id { get; set; }
+			public int CaseId { get; set; }
+		}
+
+		public class DecisionOutcomeCreatedCaseUpdatedHandler : INotificationHandler<DecisionOutcomeCreatedNotification>
+		{
+			private readonly ConcernsDbContext _context;
+			public DecisionOutcomeCreatedCaseUpdatedHandler(ConcernsDbContext context)
+			{
+				_context = context;
+			}
+
+			public async Task Handle(DecisionOutcomeCreatedNotification notification, CancellationToken cancellationToken)
+			{
+				ConcernsCase cc = await _context.ConcernsCase.SingleOrDefaultAsync(f => f.Id == notification.CaseId, cancellationToken: cancellationToken);
+                DecisionOutcome decisionOutcome = await _context.DecisionOutcomes.SingleOrDefaultAsync(f => f.DecisionOutcomeId == notification.Id, cancellationToken: cancellationToken);
+				cc.CaseLastUpdatedAt = decisionOutcome.UpdatedAt.DateTime;
```

Comment:
Should the CaseLastUpdatedAt field be set to the decisionOutcome.CreatedAt value instead of the updatedat value?